src/components/visualizations/chart/HighChartsRenderer.tsx

```diff
@@ -277,14 +280,24 @@ export default class HighChartsRenderer extends React.PureComponent<
             },
         );
 
-        const renderLegendFirst = legend.position === TOP || (legend.position === LEFT && !showFluidLegend);
+        const isLegendRenderedFirst: boolean =
+            legend.position === TOP || (legend.position === LEFT && !showFluidLegend);
 
         return (
             <div className={classes}>
-                {renderLegendFirst && this.renderLegend()}
+                {isLegendRenderedFirst && this.renderLegend()}
                 {this.renderHighcharts()}
-                {!renderLegendFirst && this.renderLegend()}
+                {!isLegendRenderedFirst && this.renderLegend()}
             </div>
         );
     }
+
+    private realignPieOrDonutChart() {
+        const {
+            chartOptions: { type },
+        } = this.props;
+        if (isPieOrDonutChart(type) && this.chartRef) {
+            alignChart(this.chartRef.getChart());
```

Comment:
consider
`alignChart(this.chartRef.getChart());`
to
```
const { getChart } = this.chartRef;
...
alignChart(getChart());
...
```

Reply:
`chartRef` could be null so it might cause error.
I fix it in other way.